### ISSUE TITLE
Bootstrapping against prebuild local dependencies fails when /usr/bin or /usr/include contain symlinks

### ISSUE THREAD

**Issue by mhthies** [just-buildsystem/justbuild#12]
I'm trying to package *justbuild* for ArchLinux, linking against native shared libraries, provided by ArchLinux packages as far as possible. Thus, I use the bootstrap.py script with the environment variables `PACKAGE=YES LOCALBASE=/usr`.

After working around the other issues, this resulted in the following problem:

```txt
Running ['/tmp/tmpxaow1ajw/bootstrap-just', 'analyse', '-C', '/tmp/tmpxaow1ajw/repo-conf.json', '-D', '{"COMPILER_FAMILY": "clang", "CC": "/usr/bin/clang", "CXX": "/usr/bin/clang++", "AR": "/usr/bin/ar", "ARCH": "x86_64"}', '--dump-graph', '/tmp/tmpxaow1ajw/graph.json', '--dump-artifacts-to-build', '/tmp/tmpxaow1ajw/to_build.json', '', ''] in '/tmp/tmpxaow1ajw/src'
INFO: Requested target is [["@","just","",""],{"AR":"/usr/bin/ar","ARCH":"x86_64","CC":"/usr/bin/clang","COMPILER_FAMILY":"clang","CXX":"/usr/bin/clang++"}]
ERROR: unsupported type for dir entry /usr/include/./pngconf.h
ERROR: While processing targets:
       While evaluating export target ["@","just","",""]:
       While evaluating install target ["@","just","","installed just"]:
       While evaluating configure target ["@","just","","just"]:
       While evaluating export target ["@","just","","exported-just"]:
       While analysing ["@","rules-just","CC","binary"] target ["@","just","src/buildtool/main","just"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/common","config"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/file_system","git_cas"]:
       While evaluating configure target ["@","just","","libgit2"]:
       While analysing ["@","rules-git2","CC","library"] target ["@","com_github_libgit2_libgit2","","git2"]:
       While analysing target ["@","com_github_libgit2_libgit2","","git2.h"] as implicit source target:
       Cannot determine source file "git2.h" in directory "" of repository "com_github_libgit2_libgit2"
ERROR: unsupported type for dir entry /usr/bin/ardour7-new_session
ERROR: While processing targets:
       While evaluating export target ["@","just","",""]:
       While evaluating install target ["@","just","","installed just"]:
       While evaluating configure target ["@","just","","just"]:
       While evaluating export target ["@","just","","exported-just"]:
       While analysing ["@","rules-just","CC","binary"] target ["@","just","src/buildtool/main","just"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/execution_api/local","garbage_collector"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/common","bazel_types"]:
       While analysing ["@","rules-just","CC/proto","service library"] target ["#","a0d338d0250537ba57c641b3c933e548bdd24d9b","1181df6b8bbe306c91444bdcd12cda0310104333"]:
       While evaluating install target ["@","com_github_grpc_grpc","src/compiler","grpc_cpp_plugin"]:
       While analysing target ["@","com_github_grpc_grpc","","bin/grpc_cpp_plugin"] as implicit source target:
       Cannot determine source file "grpc_cpp_plugin" in directory "bin" of repository "com_github_grpc_grpc"
ERROR: unsupported type for dir entry /usr/bin/ardour7-new_session
ERROR: While processing targets:
       While evaluating export target ["@","just","",""]:
       While evaluating install target ["@","just","","installed just"]:
       While evaluating configure target ["@","just","","just"]:
       While evaluating export target ["@","just","","exported-just"]:
       While analysing ["@","rules-just","CC","binary"] target ["@","just","src/buildtool/main","just"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/execution_api/local","garbage_collector"]:
       While analysing ["@","rules-just","CC","library"] target ["@","just","src/buildtool/common","bazel_types"]:
       While analysing ["@","rules-just","CC/proto","service library"] target ["#","a0d338d0250537ba57c641b3c933e548bdd24d9b","1181df6b8bbe306c91444bdcd12cda0310104333"]:
       While analysing ["@","rules-protobuf","data","staged"] target ["@","protobuf","","protoc"]:
       While analysing target ["@","protobuf","bin","protoc-3.21.12.0"] as implicit source target:
       Cannot determine source file "protoc-3.21.12.0" in directory "bin" of repository "protobuf"
```

The mentioned dir entries with unsupported types (`/usr/include/./pngconf.h`, `/usr/bin/ardour7-new_session`) are symbolic links to other files within the same directory.

I was able to work around this issue by patching *justbuild* to ignore symlinks instead of aborting directory scanning: https://github.com/mhthies/justbuild/commit/4b283e98feac0409bb5ca8feb9f3e72b17a29e01

**Comment by aehlig** [comment]
As mentioned on #14, I propose to proceed as described in the design document in #17.

**Comment by aehlig** [comment]
I believe this is fixed in 34ae76b4bfc5d7c153f6c47f5257c2db515e70f1
